### PR TITLE
Extendsystempropertycleanup

### PR DESCRIPTION
**Pull request by carstenartur**
System properties for boolean values are substituted already in the cleanup. Now add Integer and Long support.

### CODE REVIEW

File: .github/workflows/codeql-analysis.yml
Comment (jjohnstn) [inline .github/workflows/codeql-analysis.yml:1]: Hi Carsten, good to see you back.  Why are the yml files part of this issue?  They aren't part of the system property cleanup extension.  Did you perhaps mean to submit them in a separate issue/commit and happened to have them in your branch?

Comment (carstenartur) [reply .github/workflows/codeql-analysis.yml:1]: Yes, didn't want them to be part of this commit. Still need to find out how to remove them from patch in the github web gui like it was possible in gerrit.

Not sure that I'm going to contribute a lot. But at least had some spare time to look at how things are working after the migration to github.

Comment (jjohnstn) [reply .github/workflows/codeql-analysis.yml:1]: I believe one way would be to git format-patch the one commit you want, then do a git reset --hard to the patch before, then do a git am <patch you saved to the branch.  You should then be able to do a git push --force of the new version of the branch and it should show up here sans the other stuff.  Not fond of github myself.

Comment (carstenartur) [reply .github/workflows/codeql-analysis.yml]: Seems to work fine in the github web gui for the pull request. Before I was looking for this in the github app but havn't found it there.

Comment (jjohnstn) [reply .github/workflows/codeql-analysis.yml]: Hi Carsten, it seems to work fine, but I don't know how to squash your commits and we shouldn't merge adding/deleting files.  Please do a git rebase -i locally and squash all the 5 commits into one commit so the xml changes aren't even seen or mentioned.  Also the PR is marked as Draft so I can't merge it until it is moved out of Draft.  Did I mention I hate github??  I would have solved this in gerrit in a flash.

Comment (carstenartur) [reply .github/workflows/codeql-analysis.yml]: I think it is much easier than you think. It is just a question of your configuration.
https://egghead.io/lessons/github-automatically-squash-commits-on-github-when-merging-pull-requests

File: org.eclipse.jdt.ui/ui/org/eclipse/jdt/internal/ui/preferences/cleanup/JavaFeatureTabPage.java
Comment (jjohnstn) [inline org.eclipse.jdt.ui/ui/org/eclipse/jdt/internal/ui/preferences/cleanup/JavaFeatureTabPage.java:129]: You need to add an intent() call before the Long property checkbox is created as it gets pushed left-justified and makes the rest of the check-boxes look incorrect.  Other than that, it looks good.

**Review by jjohnstn — CHANGES_REQUESTED** [review]
There is a minor UI issue with the new checkboxes that is easily fixed with an added indent (actually called intent()).  See comment.

**Comment by jjohnstn** [comment]
Hi Carsten, after thinking about it, is there ever going to be a reason a user will want to fix Boolean values but not Integer or Long values?  I think it would be simpler for the user if we integrated Long and Integer support into one option (Boolean/Long/Integer).  What do you think?

**Comment by carstenartur** [comment]
> Hi Carsten, after thinking about it, is there ever going to be a reason a user will want to fix Boolean values but not Integer or Long values?  I think it would be simpler for the user if we integrated Long and Integer support into one option (Boolean/Long/Integer).  What do you think?

Fine with me.
You still have some control in the preview...

**Comment by jjohnstn** [comment]
Hi Carsten, I have taken a stab at amending your patch.  I can't figure out how to do a rebase that specifies your branch (tried many attempts - thank you github).  Anyway, to save time, I created my own branch, applied your one patch and amended it with changes.  Take a look at:
https://github.com/eclipse-jdt/eclipse.jdt.ui/pull/24  and let me know if you like it.  I changed the option to specify Boxed type instead of just Boolean.  I thought that adding Boolean/Long/Integer would have been too long for the message.  The same cleanup constant is used (with the word boolean) so existing users of the option won't be impacted.

**Comment by carstenartur** [comment]
> Hi Carsten, I have taken a stab at amending your patch.  I can't figure out how to do a rebase that specifies your branch (tried many attempts - thank you github).  Anyway, to save time, I created my own branch, applied your one patch and amended it with changes.  Take a look at:
> 
> https://github.com/eclipse-jdt/eclipse.jdt.ui/pull/24  and let me know if you like it.  I changed the option to specify Boxed type instead of just Boolean.  I thought that adding Boolean/Long/Integer would have been too long for the message.  The same cleanup constant is used (with the word boolean) so existing users of the option won't be impacted.


Sounds good to me.

Hope I will be able to check your changes out next week. But feel free to merge right now anyway if you think it is fine that way.

Regarding the github workflow I think it might make sense in future to try to move away slightly from the gerrit way to do things. You can even let github do merging automatically as soon as all checks succeed.

In any case you should have already permission to work on my pull requests as long as jdt.ui github is configured correctly.

See
https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork

Thank you so far

https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/incorporating-changes-from-a-pull-request/automatically-merging-a-pull-request

**Comment by jjohnstn** [comment]
Hi Carsten, when I attempted to push the changes git failed and said I had to specify a remote branch.  I tried a number of options and all them resulted in github said the branch didn't exist.  What should I have specified to merge the changes to your PR?

**Comment by carstenartur** [comment]
> Hi Carsten, when I attempted to push the changes git failed and said I had to specify a remote branch.  I tried a number of options and all them resulted in github said the branch didn't exist.  What should I have specified to merge the changes to your PR?

Which repository and branch did you work on? Have you seen https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/committing-changes-to-a-pull-request-branch-created-from-a-fork ? (I know, that must look strange and particular "heavy" or messy for gerrit users - but I think while egit support for this approach could be better at the same time focus on letting the contributer fix his pull request seems to be evident on github.) 
Some of the Suggestions can be directly committed within the peer review.
See

https://egghead.io/lessons/github-add-suggestions-in-a-github-pr-review

Otherwise you have to work on my branch in my fork.

**Comment by jjohnstn** [comment]
Thanks Carsten for the info.  I did not do a clone and tried using a fetch.

**Pull request closed without merging**
